update_ip/services/nfsn.py

```diff
@@ -1,64 +1,60 @@
+import logging
+
+import tld
+import pynfsn
+
 from update_ip.services.base import BaseDNSService, DNSServiceError
-try:
-    from pynfsn import pynfsn
-except ImportError:
-    raise Exception("This service requires the pynfsn package. You can find it on pypi or github")
 
-import logging
-log= logging.getLogger('update_ip.services.nfsn')
+log = logging.getLogger('update_ip.services.nfsn')
+
+ALLOW_INEXISTENT = True  # If this is false, I'll refuse to proceed if the domain record doesn't already exist
 
-ALLOW_INEXISTENT= True #If this is false, I'll refuse to proceed if the domain record doesn't already exist
 
-def split_domain(domain):
-    '''splits a complete domain into a pair of (subdomain, domain)'''
-    s= domain.split(".")
-    subdomain, domain= ".".join(s[:-2]), ".".join(s[-2:])
-    return (subdomain, domain)
-        
 class NearlyFreeSpeechService(BaseDNSService):
     name = 'NearlyfreeSpeech'
-    TTL= 300    #5 minutes of dns record ttl
-    
+    TTL = 300  # 5 minutes of dns record ttl
+
     def __init__(self, username, api_key, **kwargs):
         if not username or not api_key:
-            raise DNSServiceError('Username and api_key are required for the '
-                                 'NearlyFreeSpeech service.')
-        self.nfsn= pynfsn.NFSN(username, api_key)
+            raise DNSServiceError('Username and api_key are required for the NearlyFreeSpeech service.')
+        self.nfsn = pynfsn.NFSN(username, api_key)
 
+        # Updates the known TLD names
+        tld.utils.update_tld_names()
```

Comment:
I've added this to command options, so now you can simply do: "update-ip -t" to update TLDs, so it's more useful since you only need to update when you need to add a domain with a new TLD.

Also i've handled the exception with user permissions